src/common/Utilities/StringConvert.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_STRINGCONVERT_H
+#define TRINITY_STRINGCONVERT_H
+
+#include "Define.h"
+#include "Errors.h"
+#include "Optional.h"
+#include "Util.h"
+#include <charconv>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+namespace Trinity::Impl::StringConvertImpl
+{
+    template <typename T, typename = void> struct For
+    {
+        static_assert("Unsupported type used for ToString or StringTo");
+        /*
+        static Optional<T> FromString(std::string_view str, ...);
+        static std::string ToString(T&& val, ...);
+        */
+    };
+
+    // @todo relax this once proper std::from_chars support exists
+    template <typename T>
+    struct For<T, std::enable_if_t<std::is_integral_v<T> && !std::is_same_v<T, bool>>>
+    {
+        static Optional<T> FromString(std::string_view str, int base = 10)
+        {
+            if (base == 0)
+            {
+                if (StringEqualI(str.substr(0, 2), "0x"))
+                {
+                    base = 16;
+                    str.remove_prefix(2);
+                }
+                else if (StringEqualI(str.substr(0, 2), "0b"))
+                {
+                    base = 2;
+                    str.remove_prefix(2);
+                }
+                else if (str.substr(0, 1) == "0")
+                {
+                    base = 8;
+                    str.remove_prefix(1);
+                }
+                else
+                    base = 10;
+
+                if (str.empty())
+                    return std::nullopt;
+            }
+
+            char const* const start = str.data();
+            char const* const end = (start + str.length());
+
+            T val;
+            std::from_chars_result const res = std::from_chars(start, end, val, base);
+            if (res.ptr == end)
+                return val;
+            else
+                return std::nullopt;
+        }
+
+        static std::string ToString(T val)
+        {
+            std::string buf(20,0); /* largest unsigned 64-bit value is 20 decimal characters, largest signed 64-bit value is 19 */
```

Comment:
```suggestion
            std::string buf(20,'\0'); /* largest unsigned 64-bit value is 20 decimal characters, largest signed 64-bit value is 19 */
```

Also, comment isnt right because "longest" 64 bit signed value is also 20, dont forget the '-' character

Reply:
meant 19 without sign

Reply:
btw you missed the `0` to `'\0'` change